app/uk/gov/hmrc/selfservicetimetopay/service/CalculatorService.scala

```diff
@@ -197,7 +201,26 @@ object CalculatorService {
       monthDifference(todayDate, el)).getOrElse(maxAllowedMonthlyInstalments)
 
     val smallestDifferance = List(differenceNextSubmissionReturn, differenceEarliestDueDate).min
-    if (smallestDifferance > maxAllowedMonthlyInstalments) maxAllowedMonthlyInstalments else smallestDifferance
+    val gap = if (smallestDifferance > maxAllowedMonthlyInstalments) maxAllowedMonthlyInstalments else smallestDifferance
+
+    JourneyLogger.info(
+      s"""
+         |gap in months = $gap
+         |todayDate = $todayDate
+         |nextSubmissionReturn = $nextSubmissionReturn
+         |differenceNextSubmissionReturn = $differenceNextSubmissionReturn
+         |differenceEarliestDueDate = $differenceEarliestDueDate
+         |smallestDifferance = $smallestDifferance
+         |""".stripMargin,
+      selfAssessment
+    )
+    gap
```

Comment:
the `gap` tends to be `0` in some cases which causes bunch of exceptions. We need to debug what causes it to be `0`